src/roofs.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+#ifndef ROOFS_H
+#define ROOFS_H
+
+#include <map>
+#include <memory>
+#include "string_id.h"
+
+class map;
+class game;
+struct tripoint;
+
+struct support_cache;
+
+struct ter_t;
+using ter_id = int_id<ter_t>;
+
+enum direction : unsigned;
+
+enum class roof_dir : size_t {
```

Comment:
One from line is huge, I only want 4 directions. By using one with 4 directions I can use it as array indices too - that makes the thing more readable.